assets/js/util/sanitize.js

```diff
@@ -8,3 +8,25 @@ export const sanitizeHTML = ( unsafeHTML, domPurifyConfig = {} ) => {
 		__html: purify.sanitize( unsafeHTML, domPurifyConfig ),
 	};
 };
+
+/**
+ * Takes a URL string, removes the trailing stash if any and returns it.
+ *
+ * @since 1.18.0
+ * @private
+ *
+ * @param {string} url The URL with or without trailing slash.
+ * @return {string|null} The URL string after removing the trailing slash.
+ */
+export const unTrailingSlashIt = ( url ) => {
+	if ( typeof url !== 'string' ) {
+		return null;
+	}
+
+	if ( url.endsWith( '/' ) ) {
+		// remove the trailing slash
+		return url.slice( 0, -1 );
+	}
+
+	return url;
```

Comment:
Similar to the PHP function, I think this utility should remove any number of trailing slashes rather than assuming there is only one.

Similarly, I think this can be reduced to a simple one-liner:
```suggestion
	return string?.replace?.( /\/+$/, '' );
```

In this case, it would return `undefined` if the given input is not a string, which could have a default handled with `|| null` or `|| ''`, but I'm okay with it returning `undefined` as well since this is an unexpected value.